### PR TITLE
[No Hope] Hope for No Hope

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
Mods "[No Hope] Adds a difficulty setting prompt to determine item spawn rates"

#### Purpose of change
In it's current state No Hope is falling behind vanilla mapgen while also causing errors on alot of PRs touching existing mapgen due to No Hope's duped versions of vanilla maps that often share vanilla palettes or vice versa, adding extra work in a demotivating way after you think you're done.
This would be fine if No Hope had an active maintainer and this was the only way for the mod to work but the vast majority of changes are just to vehicles which are now overridden by an external option, and item spawn rates which Night added code to deal with but never reached it to the mod.
TLDR No Hope is currently unmaintainable

#### Describe the solution

Item category spawn rates can be overridden via EoCs.
Adds a difficulty selection EoC to game start (and load for existing saves) that sets them.
Deletes most of No Hopes altered mapgen that since being added has better alternatives. (I went through all the maps in 2 passes to check for intentional changes that aren't vehicle/item spawn rate related)
Reorganises various files including the necropolis files whose vanilla versions were just shoved in a folder rather than distributed into appropriate places.
Reenables the necropolis because whoever moved it to No Hope forgot to do so.
Changes the house palette distribution in favour of messy damaged houses.
Adds various comments explaining differences between remaining duped mapgen and their vanilla counterparts as well as adding various TODOs to the mod's files in general.

#### Describe alternatives you've considered

I'm not very happy with splitting the rates into their own struct I were just having problems otherwise, if someone smarter wants to review it that'd be welcome

#### Testing

Changed a difficulty to 0 in all the categories, got expected patchy results of some mapgen spawning no items in those categories while others using item placement methods not currently supported by the category spawn rates like "(place_)loot" spawning regular rates.
The EoC takes place early enough that the initial reality bubble is affected.

#### Additional context

I could well have missed some purposeful changes to mapgen due to the sheer volume, if you're aware of any intentional differences that I'm deleting please comment below.

The item category spawn stuff doesn't cover all our item spawn methods and needs expanding on but I don't view that as a blocker when noone's going to work on it otherwise and this problem will just linger. Outside of basic mapgen item placement methods the main two specific regressions are fuel pumps and turret ammo.

I plan on doing a few follow up PRs to give some love to No Hope to hopefully make up for the temporary regression and I'll also take a crack at expanding the spawn rate usage in the code to more relevant places.

#70522 would still be nice for the remaining maps

### CODE REVIEW

**Comment by Procyonae** [comment]
> i think no hope makes/tries to make maps more destroyed and less full of furniture, but i think it also can be resolved in a better way, like changing the weight of some pallets

For the house palette it should already be changing the variant weights but I haven't found whether it currently is or not yet otherwise I'll add it.
For damaged/looted etc map variants with their own palettes I previously added functionality so you can alter vanilla weights with mods so they won't need duped maps.
For other maps ideally we'd have vanilla damaged versions of the palettes/maps and it would do a similar thing, if I notice any well done ones I'll port them over to vanilla.

**Comment by I-am-Erk** [comment]
> i think no hope makes/tries to make maps more destroyed and less full of furniture, but i think it also can be resolved in a better way, like changing the weight of some pallets

we could possibly do it by scattering more "smash things up" map extras around the world at high frequency

**Comment by Procyonae** [comment]
> we could possibly do it by scattering more "smash things up" map extras around the world at high frequency

Eh that's really messy and we want variable damage for vanilla anyway, I'd rather it just ramps up the vanilla versions

**Comment by I-am-Erk** [comment]
> > we could possibly do it by scattering more "smash things up" map extras around the world at high frequency
> 
> Eh that's really messy and we want variable damage for vanilla anyway, I'd rather it just ramps up the vanilla versions

Well yes, that's the better solution but I thought you were fixing a mod. We prefer to do messy hacks with mods, sirrah

**Comment by MNG-cataclysm** [comment]
Is Night Pyranik no longer maintaining or working on No Hope? I thought that they were actively developing it.

**Comment by Kantonine** [comment]
Can't wait for this! The big reason why i stepped away from no-hope was the fact that the wider mapgen was screwed up, leading to incosistencies for urban and rural mapgen

**Comment by Procyonae** [comment]
> Is Night Pyranik no longer maintaining or working on No Hope? I thought that they were actively developing it.

Night's last No Hope PR was in June, there was some drama around a few of their mainline PRs being rejected since which they've mainly stopped work on CDDA. They also just generally aren't active on github atm.
#64618 (I also just realised my title coincidently plays on the first comment here ;D)

File: data/mods/No_Hope/Mapgen/bunker.json [hunk 1]
```diff
@@ -99,7 +100,8 @@
   {
     "type": "mapgen",
     "method": "json",
-    "om_terrain": [ "bunker_basement_1" ],
+    "//": "Non blood version? TODO: What else does this do differently to the one above?",
```
Comment (GuardianDll) [inline data/mods/No_Hope/Mapgen/bunker.json:103]: this one doesn't have `bunker_basement_suicide`

**Comment by Night-Pryanik** [comment]
@Procyonae I'm not involved in the mod development anymore so you are free to do whatever you think appropriate for it. If you think your changes are fine, they are fine.

**Comment by Procyonae** [comment]
Fair enough, hope you're doing well

**Comment by Night-Pryanik** [comment]
Unfortunately, I'm not doing well. Yesterday I lost my father to cancer.

**Comment by Procyonae** [comment]
> Unfortunately, I'm not doing well. Yesterday I lost my father to cancer.

My condolences, if talking to some random peep on the internet would help at any point then my discord dms are open.

File: data/mods/No_Hope/modinfo.json [hunk 2]
```diff
@@ -4,15 +4,15 @@
     "id": "no_hope",
     "name": "No Hope",
     "authors": [ "Night_Pryanik" ],
-    "maintainers": [ "Night_Pryanik" ],
```
[the remark this answers is not in the document]
Comment (Procyonae) [reply data/mods/No_Hope/modinfo.json:7]: Adding myself there would imply I'm in some way in charge of the mod's direction which I don't plan on being and that the mod is actively being worked on which it won't really be, I plan on doing the bare minimum to keep this mod in an unobsoleted state outside of potentially a few minor additions in the near future and code additions to improve the spawn rate functionality. No Hope very much still wants someone else to take the mantle and ideally push the mod to be a more unique experience which I don't plan on tackling.

[the remark this answers is not in the document]
Comment (I-am-Erk) [reply data/mods/No_Hope/modinfo.json:7]: Technically a person can fix the mod up to prevent obsoletion without accepting maintainer role.

**Pull request merged**